File: scala/src/main/scala/io/digitalstream/library/Json.scala
/**
  * Created by sebby on 10/1/16.
  */
package io.digitalstream.library

import spray.json.DefaultJsonProtocol

case class Point(x: Float, y: Float)

trait PointProtocol extends DefaultJsonProtocol {
  implicit val pointFormat = jsonFormat2(Point)
}

object Json extends PointProtocol {

  def run(): Unit = {
    val point = Point(1.0f, 2.0f)
    val serialized = pointFormat.write(point);
    println(s"serialized = ${serialized.compactPrint}")
    val deserialized = pointFormat.read(serialized)
    println(s"deserialized = $deserialized")
  }
}

File: kotlin/src/main/kotlin/io/digitalstream/library/Json.kt
/**
 * Created by sebby on 10/1/16.
 */
package io.digitalstream.library

import com.github.salomonbrys.kotson.fromJson
import com.google.gson.Gson

data class Point(val x: Float, val y: Float)

object Json {

  fun run() {
    val gson = Gson()
    val point = Point(1.0f, 2.0f)
    val serialized = gson.toJson(point)
    println("serialized = $serialized")
    val deserialized = gson.fromJson<Point>(serialized)
    println("deserialized = $deserialized")
  }
}
